server/src/main/java/org/elasticsearch/search/aggregations/pipeline/BucketHelpers.java

```diff
@@ -207,6 +208,9 @@ public static Double resolveBucketValue(
                     value = ((Number) propertyValue).doubleValue();
                 } else if (propertyValue instanceof InternalNumericMetricsAggregation.SingleValue) {
                     value = ((InternalNumericMetricsAggregation.SingleValue) propertyValue).value();
+                } else if (propertyValue instanceof NumericMetricsAggregation.MultiValue) {
+                    final String metricName = aggPathAsList.get(aggPathAsList.size() - 1);
+                    value = ((NumericMetricsAggregation.MultiValue) propertyValue).value(metricName);
```

Comment:
Looks like this breaks all other aggregations implementing `NumericMetricsAggregation.MultiValue`...

Reply:
I think I need to implement `SingleValue` not `MultiValue`.